src/service/resource.js

```diff
@@ -236,6 +239,14 @@ export class Resource {
     return this.id_;
   }
 
+  /**
+   * Returns whether the resource was ever in viewport.
+   * @return {boolean}
+   */
+  hasEnteredViewport() {
```

Comment:
Nit: `hasBeenInViewport` is a bit clearer. "Has entered" may imply that it's currently in viewport for some readers.

Reply:
bigger nit: this function is no longer needed without eer. I'm removing it